### ISSUE TITLE
Allow access to all entry properties

### ISSUE THREAD

**Issue by MarcusWichelmann** [perry-mitchell/webdav-client#134]
Hi,

is it possible to access WebDAV entry properties that are not directly exposed by the api?
I'm connecting to a server with a compilance class greater than 1 and 2 which returns some more properties that are not part of the WebDAV standard.

Example response:
```XML
    <D:propstat>
      <D:prop>
        <D:isvirtual>true</D:isvirtual>         <-- this one
        <D:resourcetype>
          <D:collection />
        </D:resourcetype>
        <D:getlastmodified>Fri, 18 Jan 2019 15:51:16 GMT</D:getlastmodified>
        <D:creationdate>2019-01-18T15:51:16.106Z</D:creationdate>
        <D:getcontentlength>0</D:getcontentlength>
        <D:lockdiscovery />
        <D:supportedlock />
      </D:prop>
      <D:status>HTTP/1.1 200 OK</D:status>
    </D:propstat>
```

Do you know a way to do this?
If not, I'd be happy to submit a pull request.

**Edit:** Doesn't seem to be possible: https://github.com/perry-mitchell/webdav-client/blob/9944d22400955702da4a8c54b385250d0bd46f1c/source/interface/dav.js#L37
Maybe this could be implemented by returning the raw `props` when `{ details: true }` is set like it's done with the headers?

**Edit 2:** When looking at NextCloud WebDAV responses you can see that they're using custom propierties, too. So this is probably not only for my a relevant feature.

Kind regards,
Marcus

**Comment by perry-mitchell** [comment]
Hi @MarcusWichelmann - That's a great suggestion. I agree with having such a feature. I think it'd be best to have it under a flag, default off, and it could even form part of the [detailed responses](https://github.com/perry-mitchell/webdav-client#detailed-responses) functionality and structures.

I'd be happy to accept a PR. If you do happen to attempt this it'd be great to have tests added as well so it's crystal what's being returned. We're using a node-based webdav server in testing, so I'm not sure what other props are returned, but hopefully it works fine. If not we can always return a fake response (XML) with some extra properties.

**Comment by MarcusWichelmann** [comment]
Great, I'll play around a bit and create a PR.

**Comment by MarcusWichelmann** [comment]
Here it is.
Would be nice if you could check it out.

**Comment by perry-mitchell** [comment]
Released in 2.3.0

**Comment by MarcusWichelmann** [comment]
Thank you very much!
